lib/ensure_content_type.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+var cType = require('content-type');
+var mwUtil = require('./mwUtil');
+
+// Utility function to split path & version suffix from a profile parameter.
+function splitProfile(profile) {
+    var match = /^(.*)\/([0-9\.]+)$/.exec(profile);
+    return {
+        path: match[1],
+        version: match[2],
+    };
+}
+
+
+/**
+ * Simple content type enforcement
+ *
+ * - Assumes that the first `produces` array entry is the latest.
+ * - Repeats request with `no-cache` header set if the content-type does not
+ *   match the latest version.
+ */
+
+function checkContentType(hyper, req, next, expectedContentType, responsePromise) {
+    return responsePromise
+    .then(function(res) {
+        if (res.headers && res.headers['content-type'] !== expectedContentType) {
+            // Parse the expected & response content type, and compare
+            // profiles.
+            var expectedProfile = cType.parse(expectedContentType).parameters.profile;
```

Comment:
When I was experimenting with the `content-type` module, I've found that `parse` is pretty slow. Since we don't expect to have a lot of different content-types, it could be cool to cache the results and key them on the `content-type` string.